crates/sgx_validation/src/lib.rs

```diff
@@ -80,38 +88,98 @@ impl Sgx {
+                            break;
+                        }
+                    }
+                    if !signed {
+                        return Err(anyhow!("untrusted enarx signer"));
+                    }
+                }
+
+                if config_ref.enarx_signer_blacklist.is_some() {
+                    let mut blacklisted = false;
+                    for signer in config_ref.enarx_signer_blacklist.as_ref().unwrap() {
+                        if rpt.mrsigner == signer.as_slice() {
+                            blacklisted = true;
+                            break;
+                        }
+                    }
+                    if blacklisted {
+                        return Err(anyhow!("untrusted enarx signer"));
+                    }
+                }
+
+                if config_ref.enarx_hash.is_some() {
+                    let mut signed = false;
+                    for hash in config_ref.enarx_hash.as_ref().unwrap() {
+                        if rpt.mrenclave == hash.as_slice() {
+                            signed = true;
+                            break;
+                        }
+                    }
+                    if !signed {
+                        return Err(anyhow!("untrusted enarx hash"));
+                    }
+                }
+
+                if config_ref.enarx_hash_blacklist.is_some() {
+                    let mut blacklisted = false;
+                    for hash in config_ref.enarx_hash_blacklist.as_ref().unwrap() {
+                        if rpt.mrenclave == hash.as_slice() {
+                            blacklisted = true;
+                            break;
+                        }
+                    }
+                    if blacklisted {
+                        return Err(anyhow!("untrusted enarx hash"));
+                    }
+                }
+
+                if config_ref.cpu_svn.is_some() {
+                    let conf_version = config_ref.cpu_svn.as_ref().unwrap();
+                    for index in 0..rpt.cpusvn.len() {
+                        if rpt.cpusvn.get(index).unwrap() < conf_version.get(index).unwrap() {
+                            return Err(anyhow!("untrusted cpu"));
+                        }
+                    }
+                }
+
+                if config_ref.enclave_product_id.is_some()
+                    && rpt.enclave_product_id() != config_ref.enclave_product_id.unwrap()
+                {
+                    return Err(anyhow!("untrusted enclave product id"));
+                }
+
+                if config_ref.enclave_security_version.is_some()
+                    && rpt.enclave_security_version() < config_ref.enclave_security_version.unwrap()
+                {
+                    return Err(anyhow!("untrusted enclave"));
+                }
+
+                if rpt
+                    .attributes()
+                    .features()
+                    .intersects(config_ref.features())
```

Comment:
shouldn't this be
```rust
                if rpt.attributes().features().difference(config_ref.features())
```

Reply:
That use of `difference()` returns a value that I don't know how to parse. Also the area of how it differs isn't important, just need to know if the bits match.

Reply:
I'm not sure what to do here, I haven't been able to get a desired outcome. @rvolosatovs

Reply:
Fixed.